tests/system-tests/rdscore/internal/rdscorecommon/sriov-pod-level-bond.go

```diff
@@ -0,0 +1,1197 @@
+
+		return fmt.Errorf("failed to run command from within pod %q in namespace %q: %w",
+			clientPod.Definition.Name, clientPod.Definition.Namespace, err)
+	}
+
+	glog.V(100).Infof("Change pod-level bond interface %s for the pod %s in namespace %s state to the %s",
+		interfaceName, clientPod.Definition.Name, clientPod.Definition.Namespace, expectedInfState)
+
+	cmdToRun = []string{"bash", "-c", "ip link show up"}
+
+	glog.V(100).Infof("Execute command: %q", cmdToRun)
+
+	err = wait.PollUntilContextTimeout(
+		context.TODO(),
+		time.Second*5,
+		time.Minute*1,
+		true,
+		func(ctx context.Context) (bool, error) {
+			output, err = clientPod.ExecCommand(cmdToRun, clientPod.Object.Spec.Containers[0].Name)
+
+			if err != nil {
+				glog.V(100).Infof("Error running command from within a pod %q in namespace %q: %v",
+					clientPod.Definition.Name, clientPod.Definition.Namespace, err)
+
+				return false, nil
+			}
+
+			glog.V(100).Infof("Successfully executed command from within a pod %q in namespace %q",
+				clientPod.Definition.Name, clientPod.Definition.Namespace)
+
+			result = output.String()
+
+			glog.V(100).Infof("Command's output:\n\t%v", result)
+
+			if toDisable && strings.Contains(result, fmt.Sprintf("%s:", interfaceName)) {
+				glog.V(100).Infof("interface %q not in the state %q", interfaceName, expectedInfState)
+
+				return false, nil
+			}
+
+			if !toDisable && !strings.Contains(result, fmt.Sprintf("%s:", interfaceName)) {
+				glog.V(100).Infof("interface %q not in the state %q", interfaceName, expectedInfState)
+
+				return false, nil
+			}
+
+			return true, nil
+		})
+
+	if err != nil {
+		glog.V(100).Infof("Failed to run command from within pod %q in namespace %q: %v",
+			clientPod.Definition.Name, clientPod.Definition.Namespace, err)
+
+		return fmt.Errorf("failed to run command from within pod %q in namespace %q: %w",
+			clientPod.Definition.Name, clientPod.Definition.Namespace, err)
+	}
+
+	return nil
+}
+
+func inspectPodLevelBondedInterfaceConfig(podObj *pod.Builder, ipv4Addr, ipv6Addr string) (bool, error) {
+	glog.V(100).Infof("Verify pod-level bonded interface configuration for pod %q in namespace %q",
+		podObj.Definition.Name, podObj.Definition.Namespace)
+
+	cmdToRun := []string{"bash", "-c", "ip a show type bond"}
+
+	var output string
+
+	glog.V(100).Infof("Execute command: %q", cmdToRun)
+
+	err := wait.PollUntilContextTimeout(
+		context.TODO(),
+		time.Second*5,
+		time.Minute*1,
+		true,
+		func(ctx context.Context) (bool, error) {
+			result, err := podObj.ExecCommand(cmdToRun, podObj.Object.Spec.Containers[0].Name)
+
+			if err != nil {
+				glog.V(100).Infof("Error running command from within a pod %q: %v",
+					podObj.Object.Name, err)
+
+				return false, nil
+			}
+
+			glog.V(100).Infof("Successfully executed command from within a pod %q in namespace %q",
+				podObj.Object.Name, podObj.Definition.Namespace)
+
+			output = result.String()
```

Comment:
while running it in my setup hit an issue that output was empty(more likely some timing issue) cause later IP as there, therefore wdyt about retrying the command if the output it empty?

Reply:
let me check; if you ran it on jenkins, please share this failure to see exact the scenario

Reply:
I did not succeed to reproduce such a failure, but I added a verification for the empty output, I believe it should solve the problem. let me know if you see it does not, thank you!